### ISSUE TITLE
Support for section in tags

### ISSUE THREAD

**Issue by tai2** [avh4/elm-format#407]
In typical elm apps,  we have a large union type for global messages.
So we need grouping these messages using comments.
But elm-format doesn't seem to recognize this kind of commenting style.

What do you think about this?

## Union Type Declarations

Expected:

```
type Msg =
    = Msg1
    | Msg2

    -- This comment introduces a section in the tags
    | Msg3
    | Msg4
````

Actual:

```
type Msg =
    = Msg1
    | Msg2

      -- This comment introduces a section in the tags
    | Msg3
    | Msg4
````

**Comment by basile-henry** [comment]
@tai2 It looks like twice the same thing. Also did you mean to have two `=` signs?

I have just tried to run your example (minus one of the `=` signs) in `elm-format` and it remove the space above the comment. Am I understanding this correctly:

Expected:

``` elm
type Msg
    = Msg1
    | Msg2

    -- This comment introduces a section in the tags
    | Msg3
    | Msg4
```

Actual:

``` elm
type Msg
    = Msg1
    | Msg2
    -- This comment introduces a section in the tags
    | Msg3
    | Msg4
```

**Comment by tai2** [comment]
Oh… What I meant is just you wrote.

Your result is different from mine, ridiculously.
My version is 

```
$ elm-format
elm-format-0.18 0.6.1-alpha
```

I'll ask other people to run the program and see results.
Please wait a day or so.

**Comment by tai2** [comment]
My project member tested the code and got same result. (elm-format-0.18 0.5.2-alpha)

@basile-henry What version do you use?

**Comment by basile-henry** [comment]
@tai2 You are right, I was using an older version:

``` sh
elm-format-0.18 0.5.2-alpha
```

However, I have just tried again with a build from the latest master:

``` sh
elm-format-0.18 0.7.0-exp-16-gd6312e45
```

And I get the exact same result as before. :smile: 

So maybe your issue has been party fixed on the current master?

Hope this helps!

**Comment by tai2** [comment]
@basile-henry really? I installed elm-format-0.18 0.7.0-exp and got the same result which contains unexpected extra space before comments.

**Comment by basile-henry** [comment]
You are right, there are 2 extra spaces in front of the comment but no extra line before the comment. I will update my previous message to take into account the extra 2 spaces that are missing.

**Comment by tai2** [comment]
Yeah. Thanks to confirm that the result is reproducible.

Now the problem is this style is preferable one or not?

**Comment by rtfeldman** [comment]
@tai2 I think your suggestion is a good one, for the reasons you outlined! I think this is the most useful way to do comments in union types, and it makes sense to change how they are indented so that it's optimized for the typical use case.
